remote-api/src/main/java/org/alfresco/rest/api/impl/TagsImpl.java

```diff
@@ -210,7 +217,7 @@ public NodeRef validateTag(String tagId)
     public NodeRef validateTag(StoreRef storeRef, String tagId)
     {
     	NodeRef tagNodeRef = nodes.validateNode(storeRef, tagId);
-    	if(tagNodeRef == null)
+    	if(tagNodeRef == null || !nodeService.getPrimaryParent(tagNodeRef).getParentRef().equals(tagParentNodeRef))
```

Comment:
We could refactor out a new private method to remove the duplication in these two methods.